common/src/test/java/com/google/auto/common/BasicAnnotationProcessorTest.java

```diff
@@ -341,23 +437,63 @@ public void properlyDefersProcessing_argumentElement() {
                 + " int myInt)",
             "  {}",
             "}");
-    JavaFileObject classBFileObject =
+    JavaFileObject generatesCodeFileObject =
         JavaFileObjects.forSourceLines(
             "test.ClassB",
             "package test;",
             "",
             "public class ClassB {",
             "  public void myMethod(@" + GeneratesCode.class.getCanonicalName() + " int myInt) {}",
             "}");
-    RequiresGeneratedCodeProcessor requiresGeneratedCodeProcessor =
-        new RequiresGeneratedCodeProcessor();
-    Compilation compilation =
-        javac()
-            .withProcessors(requiresGeneratedCodeProcessor, new GeneratesCodeProcessor())
-            .compile(classAFileObject, classBFileObject);
+    requiresGeneratedCodeDeferralTest(dependentTestFileObject, generatesCodeFileObject);
+  }
+
+  @Test
+  public void properlyDefersProcessing_recordComponent() {
```

Comment:
FYI we're discovering that these tests aren't quite complete. This is testing that a record will be deferred until the next round if it contains a component whose type is not yet defined. However what's really new with this PR is the notion of rejecting elements in a round _even if they don't reference undefined types_. So I think there need to be test methods for that too. If I add them I discover some issues with the code, because the various `FooElementBlueprint` classes weren't actually being tested.

Reply:
You are absolutely right, the testing is subpar. However, I am not sure I understand the potential problem here. I look forward to your tests. If I could do anything please don't hesitate to ask.